### PR TITLE
ite-backlight: init at v1.1

### PR DESCRIPTION
**Pull request by hexagonal-sun**
###### Motivation for this change
Add a package for `ite-backlight` utilities.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/ite-backlight/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  nativeBuildInputs = [
+    ninja
+    pkg-config
+    meson
+    cpp-gsl
+    boost
+  ];
+
+  buildInputs = [
+    libusb
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:32]: `libusb` is an alias of `libusb1`.

```suggestion
    libusb1
```

File: pkgs/misc/ite-backlight/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{ lib
+, pkgs
+, stdenv
+, ninja
+, libusb
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:5]:
```suggestion
, libusb1
```

File: pkgs/misc/ite-backlight/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{ lib
+, pkgs
+, stdenv
+, ninja
+, libusb1
+, meson
+, boost
+, fetchFromGitHub
+, pkg-config
+, cpp-gsl }:
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:10]:
```suggestion
, microsoft_gsl
}:
```

File: pkgs/misc/ite-backlight/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:1hany4bn93mac9qyz97r1l858d48zdvvmn3mabzr3441ivqr9j0a";
+  };
+
+  nativeBuildInputs = [
+    ninja
+    pkg-config
+    meson
+    cpp-gsl
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:27]:
```suggestion
    microsoft_gsl
```

File: pkgs/misc/ite-backlight/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+, cpp-gsl }:
+
+stdenv.mkDerivation rec {
+  pname = "ite-backlight";
+  version = "v1.1";
+
+  src = fetchFromGitHub {
+    owner = "hexagonal-sun";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/misc/ite-backlight/default.nix:19]:
```suggestion
  version = "1.1";

  src = fetchFromGitHub {
    owner = "hexagonal-sun";
    repo = pname;
    rev = "v${version}";
```

File: pkgs/misc/ite-backlight/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    sha256 = "sha256:1hany4bn93mac9qyz97r1l858d48zdvvmn3mabzr3441ivqr9j0a";
+  };
+
+  nativeBuildInputs = [
+    ninja
+    pkg-config
+    meson
+    cpp-gsl
+    boost
```
Comment (SuperSandro2000) [inline pkgs/misc/ite-backlight/default.nix:28]: Sure this belongs not into buildInputs?

Comment (hexagonal-sun) [reply pkgs/misc/ite-backlight/default.nix:28]: Yep; `boost` is only needed for it's `string` template library.  I believe all the code gets instantiated at compile time so there is no run-time dependency.

File: pkgs/misc/ite-backlight/default.nix [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "ite-backlight";
+  version = "1.1";
+
+  src = fetchFromGitHub {
+    owner = "hexagonal-sun";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:1hany4bn93mac9qyz97r1l858d48zdvvmn3mabzr3441ivqr9j0a";
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:21]:
```suggestion
    sha256 = "1hany4bn93mac9qyz97r1l858d48zdvvmn3mabzr3441ivqr9j0a";
```

File: pkgs/misc/ite-backlight/default.nix [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  nativeBuildInputs = [
+    ninja
+    pkg-config
+    meson
+    cpp-gsl
```
Comment (bobby285271) [inline pkgs/misc/ite-backlight/default.nix:28]: I think you forgot to update this :-) `cpp-gsl` is alias of `microsoft_gsl`.

```suggestion
    microsoft_gsl
```

Comment (hexagonal-sun) [reply pkgs/misc/ite-backlight/default.nix]: Whoops, missed that.  Should be fixed now!  Thanks!

File: pkgs/misc/ite-backlight/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/misc/ite-backlight/default.nix:29]: buildInputs is not about runtime dependencies. It is used for cross compiling and I think boost and all other libraries need to go buildInputs.

Comment (hexagonal-sun) [reply pkgs/misc/ite-backlight/default.nix]: Agreed.  Moved to `buildInputs`.

**Review by cole-h — APPROVED** [review]
Diff LGTM, `nixpkgs-review` was happy, `ite-flash --help` worked.

**Pull request merged**